mapogcsld.c

```diff
@@ -4993,8 +4996,8 @@ char *msSLDParseLogicalExpression(char *pszExpression, const char *pszWfsFilter)
 
       free(pszFLTExpression);
       pszFLTExpression = pszTmp;
+      FLTFreeFilterEncodingNode(psNode);
```

Comment:
this should be moved just after the } of line 5000, so that psNode is also freed when != NULL but when pszFLTExpression is NULL